File: 백준/Gold/11404. 플로이드/플로이드.cc
#include <stdio.h>

int main(){
    int a,bus,b,c,d;
    int cost[101][101];
    scanf("%d",&a);
    for (int i = 1; i <= a; i++) {
        for (int j = 1; j <= a; j++) {
            cost[i][j] = 2147483600;
        }
    }
    scanf("%d",&bus);
    for (int i = 1; i <= bus; i++) {
        scanf("%d %d %d",&b,&c,&d);
        if (cost[b][c] > d) cost[b][c] = d;
    }
    for (int i = 1; i <= a; i++) {
        cost[i][i] = 0;
    }
    for (int k = 1; k <= a; k++) {
        for (int i = 1; i <= a; i++) {
            for (int j = 1; j <= a; j++) {
                if (cost[i][k] != 2147483600 && cost[k][j] != 2147483600) {
                    if (cost[i][j] > cost[i][k] + cost[k][j]) {
                        cost[i][j] = cost[i][k] + cost[k][j];
                    }
                }

            }
        }
    }
    for (int i = 1; i <= a; i++) {
        for (int j = 1; j <= a; j++) {
            if (cost[i][j] == 2147483600) printf("0 "); // 연결 안 된 경우
            else printf("%d ", cost[i][j]);
        }
        printf("\n");
    }
    return 0;
}

File: 백준/Gold/11404. 플로이드/플로이드.py
a = int(input())
bus = int(input())
cost = [[float('inf')] * (a+1) for _ in range(a+1)]

for _ in range(bus):
    b,d,c = map(int, input().split())
    cost[b][d] = min(cost[b][d], c)

for k in range(1,a+1):
    for i in range(1,a+1):
        for j in range(1,a+1):
            if i == j:
                cost[i][j] = 0
            cost[i][j] = min(cost[i][j], cost[i][k] + cost[k][j])
for i in range(1,a+1):
    for j in range(1,a+1):
        if cost[i][j] == float('inf'):
            cost[i][j] = 0
        print(cost[i][j], end=' ')
    print()